### PR TITLE
#369: Made CustomTestKit scan resources in src/main/resources, too.

### PR DESCRIPTION
**Pull request by akara**
We now need to avoid resource duplication. So built dedup logic into UnicomplexBoot resource scanning.

Thanks for your pull request.  Please review the following guidelines.

- [X] Title includes issue id.
- [X] Description of the change added.
- [X] Commits are squashed.
- [X] Tests added.
- [X] Documentation added/updated.
- [X] Also please review [CONTRIBUTING.md](https://github.com/paypal/squbs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by akara** [comment]
Note: There is code provisioned for cases where the `ClassLoader` may return resource path strings other than expected or testable. So these edge cases (2 lines) are not covered. They cannot be covered unless we do something weird with the `ClassLoader` or come up with our own weird `ClassLoader` just for this test.

**Comment by Harikiranvuyyuru** [comment]
LGTM

File: squbs-testkit/src/main/scala/org/squbs/testkit/CustomTestKit.scala [hunk 1]
```diff
@@ -47,18 +50,41 @@ object CustomTestKit {
...
       .replace('.', '-')
       .replace('_', '-')
       .filter(_ != '$')
 
-  val defaultResource = Seq(getClass.getClassLoader.getResource("").getPath + "/META-INF/squbs-meta.conf")
+  /**
+    * Detects default resources for this test. These are usually at two locations:
+    * <ul>
+    *   <li>$project-path/target/scala-2.11/classes/META-INF/squbs-meta.conf</li>
+    *   <li>$project-path/target/scala-2.11/test-classes/META-INF/squbs-meta.conf</li>
```
Comment (anilgursel) [inline squbs-testkit/src/main/scala/org/squbs/testkit/CustomTestKit.scala]: Should we update the comment not to have `2.11`?  This will be invalid in the next release.
